File: 01.Python/0724.py
from typing import List

class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        totalSum = sum(nums)
        left = 0

        for i in range(len(nums)):
            right = totalSum - left - nums[i]

            if left == right:
                return i

            left += nums[i]
            
        
        return -1

File: 03.CPP/0724.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int pivotIndex(std::vector<int>& nums) {
        int totalSum = 0;
        for (int i = 0; i < nums.size(); i++) {
            totalSum += nums[i];
        }

        int left = 0;
        for (int i = 0; i < nums.size(); i++) {
            int right = totalSum - left - nums[i];

            if (left == right) {
                return i;
            }

            left += nums[i];
        }

        return -1;
    }
};